File: server/src/game/tools/conf/output/ts/PropertyConfig.ts
class PropertyConfig {
    private static _dic:Dictionary<PropertyConfig> = new Dictionary<PropertyConfig>();
    private static _arr:Array<PropertyConfig> = new Array<PropertyConfig>();

	 public Id :number ;
    
     public Type :number ;
     public Image :string ;
     public Icon :string ;
     public IconScale :number ;
     public Deviation : Array<number>  =new Array<number>();
     public Sort :number ;
     public Location : Array<number>  =new Array<number>();
    

    
    public constructor(){

    }


    public static get Dic():Dictionary<PropertyConfig>{
        return  PropertyConfig._dic;
    }
    public static get Arr():Array<PropertyConfig>{
        return PropertyConfig._arr;
    }
    public static parse(json:any[]){

        for(var i = 0; i < json.length; i++){
            let config:PropertyConfig = <PropertyConfig>json[i];
            PropertyConfig._arr.push(config);
            PropertyConfig._dic.add(config.Id,config);
        }
    }
}

File: server/src/game/tools/conf/output/go/PropertyConfig.go
package conf

import (
	//"bufio"
	"encoding/json"
	"fmt"
	//"io"
	"io/ioutil"
	"os"
	//"strings"
)

type PropertyConfigElem struct {
	Id int
	Name string
	Type int
	Image string
	Icon string
	IconScale float32
	Deviation []float32
	Sort int
	Location []int
	
}

type PropertyConfig0 struct {
	Dic map[int]*PropertyConfigElem
	Arr []*PropertyConfigElem
}

var (
	PropertyConfig PropertyConfig0
)

func loadPropertyConfig(path string) {
	file, err := os.Open(path)
	defer func() {
		if file != nil {
			file.Close()
		}
	}()
	if err != nil {
		panic(fmt.Sprint("load config error, path=", path))
	}

	PropertyConfig.Dic = make(map[int]*PropertyConfigElem)
	PropertyConfig.Arr = make([]*PropertyConfigElem, 0, 32)

	bt,err  :=  ioutil.ReadAll(file)
	json.Unmarshal(bt, &PropertyConfig.Arr)
	for _,conf := range PropertyConfig.Arr{
		PropertyConfig.Dic[conf.Id] = conf
	}
	
}
